src/radical/pilot/utils/prof_utils.py

```diff
@@ -1236,5 +1236,160 @@ def _get_unit_consumption(session, unit, udurations=None):
+        for m in task_transitions:
+            if m[0] == spec[0]:
+                break
+            m = None
+        if m:
+            assert(m[2] is None)
+            m[2] = metric
+        else:
+            task_transitions.append([spec[0], None, metric])
+
+
+        for m in task_transitions:
+            if m[0] == spec[1]:
+                break
+            m = None
+        if m:
+            assert(m[1] is None)
+            m[1] = metric
+        else:
+            task_transitions.append([spec[1], metric, None])
+
+    for t in task_transitions:
+        if t[1] is None: t[1] = 'idle'
+        if t[2] is None: t[2] = 'idle'
+
+
+    # do the same for the pilot metrics / transitions, only that now we
+    # translate `None` as `system`, which is where the pilot obtains it's
+    # resources from.  Also note that some transition events can have double
+    # entries, for example, if a event passes some resources to a sub-agent, and
+    # some other resources remain in the pilot.
+    #
+    # Note that we also handle the `agent` metrics to cover resources used by
+    # the agent (and thus not making it into usable).
+    pilot_transitions = list()
+    for metric,spec in pilot_metrics['consume'].items():
+
+        m = None
+        for m in pilot_transitions:
+            if m[0] == spec[0]:
+                break
+            m = None
+        if m:
+            assert(m[2] is None)
+            m[2] = metric
+        else:
+            pilot_transitions.append([spec[0], None, metric])
+
+
+        for m in pilot_transitions:
+            if m[0] == spec[1]:
+                break
+            m = None
+        if m:
+            assert(m[1] is None)
+            m[1] = metric
+        else:
+            pilot_transitions.append([spec[1], metric, None])
+
+    for t in pilot_transitions:
+        # agent resources from from the pilot, pilot resources from the system
+        if t[1] is None:
+            if t[2] == 'agent': t[1] = 'setup_1'
+            else              : t[1] = 'system'
+        if t[2] is None:
+            if t[1] == 'agent': t[2] = 'term'
+            else              : t[2] = 'system'
+
+    return pilot_transitions, task_transitions
+
+
+# ------------------------------------------------------------------------------
+#
+def get_resource_timelines(session, task, transitions):
+    '''
+    For each the specific task, return a set of tuples of the form:
+
+        [start, stop, metric]
+
+    which reports during time what metric has been used.
+    '''
+
+    # we need to know what pilot the task ran on.  If we don't find a designated
+    # pilot, no resources were consumed
+    tid = task.uid
+    pid = task.cfg['pilot']
+
+    if not pid:
+        # task was never assigned to a pilot
+        return dict()
+
+    if 'slots' not in task.cfg:
+        # the task was never scheduled
+        return dict()
+
+    # we heuristically switch between PRTE event traces and normal (fork) event
+    # traces
```

Comment:
Not sure I understand that comment

Reply:
Yeah, that didn't make sense - removed (cut'n'paste error)